### DISCUSSION TITLE
How to intercept XHR (Http responses/requests) using puppeteer

### DISCUSSION THREAD

**Discussion by lofti198** [hardkoded/puppeteer-sharp#2676] [Q&A]
I want to intercept XHR to generate further direct requests to the server like on screenshot attached. So I need to intercept cookies and headers of requests and responses. Is there any way to do this out of the box with Puppeteer sharp?

![image](https://github.com/hardkoded/puppeteer-sharp/assets/42960488/e0a13fb8-c6ed-4879-ab6f-3c47c79034e6)

**Comment by kblok** [comment]
@lofti198 what did you try so far?
